### PR TITLE
[bitnami/odoo] Chart standardized

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juanariza@vmware.com>

**Description of the change**

This PR standardizes the `bitnami/odoo` chart to be in line with the rest of the catalog.

**Benefits**

Standardization.

**Possible drawbacks**

None. No major bump is required since breaking changes were added support for deprecated values using `coalesce`.

**Applicable issues**

N/A

**Additional information**

N/A

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using (readme-generator-for-helm)[https://github.com/bitnami-labs/readme-generator-for-helm]
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/odoo/templates/deployment.yaml [hunk 1]
```diff
@@ -188,46 +202,33 @@ spec:
...
-            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
-            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
-            successThreshold: {{ .Values.readinessProbe.successThreshold }}
-            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
           {{- else if .Values.customReadinessProbe }}
           readinessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.customReadinessProbe "context" $) | nindent 12 }}
           {{- end }}
           {{- if .Values.startupProbe.enabled }}
-          startupProbe:
+          startupProbe: {{- omit .Values.startupProbe "enabled" "path" | toYaml | nindent 12 }}
```
Comment (migruiz4) [inline bitnami/odoo/templates/deployment.yaml:224]:
```suggestion
          startupProbe: {{- include "common.tplvalues.render" (dict "value" (omit .Values.startupProbe "enabled" "path") "context" $) | nindent 12 }}
```

File: bitnami/odoo/templates/deployment.yaml [hunk 2]
```diff
@@ -248,7 +249,7 @@ spec:
         - name: odoo-data
           {{- if .Values.persistence.enabled }}
           persistentVolumeClaim:
-            claimName: {{ .Values.persistence.existingClaim | default (include "odoo.fullname" .) }}
+            claimName: {{ (tpl .Values.persistence.existingClaim $) | default (include "common.names.fullname" .) }}
```
Comment (migruiz4) [inline bitnami/odoo/templates/deployment.yaml:252]: Does it make sense to use `|` ?
```suggestion
            claimName: {{ default (include "common.names.fullname" .) (tpl .Values.persistence.existingClaim $) }}
```

File: bitnami/odoo/templates/svc.yaml [hunk 3]
```diff
@@ -1,42 +1,43 @@
...
   {{- end }}
   {{- if (or (eq .Values.service.type "LoadBalancer") (eq .Values.service.type "NodePort")) }}
   externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy | quote }}
   {{- end }}
-  {{- if (and (eq .Values.service.type "LoadBalancer") .Values.service.loadBalancerSourceRanges) }}
-  loadBalancerSourceRanges: {{- toYaml . | nindent 4 }}
+  {{- if eq .Values.service.type "LoadBalancer" }}
+  {{- with .Values.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges: {{ .Values.service.loadBalancerSourceRanges }}
   {{- end }}
```
Comment (migruiz4) [inline bitnami/odoo/templates/svc.yaml:29]: Why use `with` here? So far we have been using:

```
  {{- if and (eq .Values.service.type "LoadBalancer") .Values.service.loadBalancerSourceRanges }}
  loadBalancerSourceRanges: {{- toYaml .Values.service.loadBalancerSourceRanges | nindent 4 }}
  {{- end }}
```

File: bitnami/odoo/templates/deployment.yaml (hunk 2 above)
Comment (juan131) [reply bitnami/odoo/templates/deployment.yaml:252]: It's exactly the same thing

File: bitnami/odoo/templates/svc.yaml [hunk 4]
```diff
@@ -1,42 +1,43 @@
...
-  {{- if (and (eq .Values.service.type "LoadBalancer") (not (empty .Values.service.loadBalancerIP))) }}
+  {{- if not (empty .Values.service.loadBalancerIP) }}
   loadBalancerIP: {{ .Values.service.loadBalancerIP }}
   {{- end }}
```
[the remark this answers is not in the document]
Comment (migruiz4) [reply bitnami/odoo/templates/svc.yaml:32]: Oh, okay, I hadn't noticed that we are using the same if for `loadBalancerIP` and `loadBalancerSourceRanges`.

File: bitnami/odoo/templates/deployment.yaml (hunk 2 above)
Comment (migruiz4) [reply bitnami/odoo/templates/deployment.yaml:252]: I know, not a question about functionality but about style. I don't have a strong opinion about any of them as far as they work.

**Review by migruiz4 — APPROVED** [review]

**Pull request merged**